src/lib/y2network/autoinst/interfaces_reader.rb

```diff
@@ -88,56 +87,81 @@ def load_generic(config, interface_section)
+          if !interface_section.ipaddr && !interfaces_section.aliases
             msg = "Configuration for #{config.name} is invalid #{interface_section.inspect}"
             raise ArgumentError, msg
           end
 
-          if !interface_section.ipaddr.empty?
-            ipaddr = IPAddress.from_string(interface_section.ipaddr)
-          end
-          # Assign first netmask, as prefixlen has precedence so it will overwrite it
-          ipaddr.netmask = interface_section.netmask if !interface_section.netmask.to_s.empty?
-          if !interface_section.prefixlen.to_s.empty?
-            ipaddr.prefix = interface_section.prefixlen.to_i
-          end
-          if !interface_section.broadcast.empty?
-            broadcast = IPAddress.new(interface_section.broadcast)
-          end
-          if !interface_section.remote_ipaddr.empty?
-            remote = IPAddress.new(interface_section.remote_ipaddr)
-          end
-          config.ip = ConnectionConfig::IPConfig.new(
-            ipaddr, broadcast: broadcast, remote_address: remote
-          )
+          config.ip = load_ipaddr(interface_section)
         end
 
         # handle aliases
-        interface_section.aliases.each_value do |alias_h|
-          ipaddr = IPAddress.from_string(alias_h["IPADDR"])
-          # Assign first netmask, as prefixlen has precedence so it will overwrite it
-          ipaddr.netmask = alias_h["NETMASK"] if alias_h["NETMASK"]
-          ipaddr.prefix = alias_h["PREFIXLEN"].delete("/").to_i if alias_h["PREFIXLEN"]
-          config.ip_aliases << ConnectionConfig::IPConfig.new(ipaddr, label: alias_h["LABEL"])
+        interface_section.aliases.values.each_with_index do |alias_h, index|
+          config.ip_aliases << load_alias(alias_h, id: "_#{index}")
         end
+
+        # startmode
         if !interface_section.startmode.to_s.empty?
           config.startmode = Startmode.create(interface_section.startmode)
         end
         if config.startmode.name == "ifplugd" && !interface_section.ifplugd_priority.to_s.empty?
           config.startmode.priority = interface_section.ifplugd_priority
         end
+
+        # mtu
         config.mtu = interface_section.mtu.to_i if !interface_section.mtu.to_s.empty?
+
+        # ethtool options
         if interface_section.ethtool_options
           config.ethtool_options = interface_section.ethtool_options
         end
+
+        # fw zone
         config.firewall_zone = interface_section.zone if !interface_section.zone.to_s.empty?
+
+        # DHCLIENT_SETHOSTNAME setup
         if !interface_section.dhclient_set_hostname.empty?
           config.dhclient_set_hostname = interface_section.dhclient_set_hostname == "yes"
         end
 
         config
       end
 
+      # Loads and intializates interface_section's ipaddr attribute
+      #
+      # @param section [Hash] hash of AY profile's interface section as obtained from parser
+      #
+      # @return [ConnectionConfig::IPConfig] created ipaddr object
```

Comment:
miss @param documentation

Reply:
as in many other methods in the file ;-) will add.